integration/commands_diagnostic_test.go

```diff
@@ -406,3 +409,130 @@ func TestCommandsDiagnosticWhatsMyURI(t *testing.T) {
 		assert.NotEqual(t, ports[0], ports[1])
 	}
 }
+
+// TestCommandWhatsMyURIConnection tests that a client uses the same connection for
+// all commands run by the client, and different clients use different connection.
+// The port number is used to validate if the same connection is used or not,
+// since unique port number is assigned to each client in the test setup.
+func TestCommandWhatsMyURIConnection(t *testing.T) {
+	t.Parallel()
+
+	// set 1 to ensure only one pool exists duration of the test,
+	// which forces a client to use a single pool
+	q1 := url.Values{}
+	q1.Set("maxPoolSize", "1")
+	q1.Set("minPoolSize", "1")
+
+	s := setup.SetupWithOpts(t, &setup.SetupOpts{
+		ExtraOptions: q1,
+	})
+
+	collection1 := s.Collection
+	databaseName := s.Collection.Database().Name()
+	collectionName := s.Collection.Name()
+
+	t.Run("SameClient", func(t *testing.T) {
+		setup.SkipExceptMongoDB(t, "https://github.com/FerretDB/FerretDB/issues/2906")
```

Comment:
Fails on FerretDB, see https://github.com/FerretDB/FerretDB/actions/runs/5373306151/jobs/9747542079?pr=2878

Reply:
All MongoDB URI query parameters are driver / client options, not MongoDB / FerretDB server options. `maxPoolSize`, `minPoolSize` and few other options speicify how many connections the client opens and when. Those options are not visible for the server.

Reply:
This was totally my fault, I forgot to add pool size to newly created listeners :sweat: